test/dataset/test_collation.py

```diff
@@ -58,3 +70,145 @@ def test_collate_feature_padding():
 
     assert features.shape[1] == correct_pad
     assert max(features_lens).item() == correct_pad
+
+
+def test_collate_custom_array():
+    EMBEDDING_SIZE = 300
+
+    cuts = CutSet.from_json("test/fixtures/ljspeech/cuts.json")
+    with NamedTemporaryFile(suffix=".h5") as f, NumpyHdf5Writer(f.name) as writer:
+        expected_xvectors = []
+        for cut in cuts:
+            expected_xvectors.append(np.random.randn(EMBEDDING_SIZE).astype(np.float32))
+            cut.xvector = writer.store_array(cut.id, expected_xvectors[-1])
+
+        xvectors = collate_custom_field(cuts, "xvector")
+        assert isinstance(xvectors, torch.Tensor)
+        assert xvectors.dtype == torch.float32
+        assert xvectors.shape == (len(cuts), EMBEDDING_SIZE)
+        for idx, xvec in enumerate(expected_xvectors):
+            torch.testing.assert_allclose(xvectors[idx], xvec)
```

Comment:
So is the idea that we would then write out the CutSet again with the new metadata included, after writing out the data?

Reply:
Yes, I believe so.

I have used it in
https://github.com/k2-fsa/icefall/blob/62ada37d4ec0205512264f8c173b33691309ec29/egs/librispeech/ASR/conformer_ctc/ali.py#L230

at https://github.com/k2-fsa/icefall/pull/125

Reply:
That's right.